clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -94,6 +96,36 @@ def SInt16 : SInt<16>;
 def SInt32 : SInt<32>;
 def SInt64 : SInt<64>;
 
+//===----------------------------------------------------------------------===//
+// FloatType
+//===----------------------------------------------------------------------===//
+
+class CIR_FloatType<string name, string mnemonic> 
+    : CIR_Type<name, mnemonic,
+               [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>],
+               "::mlir::cir::FloatType"> {}
+
+def CIR_Single : CIR_FloatType<"Single", "float"> {
+  let summary = "CIR single-precision float type";
+  let description = [{
+    Floating-point type that represents the `float` type in C/C++. Its
+    underlying floating-point format is the IEEE-754 binary32 format.
+  }];
+}
+
+def CIR_Double : CIR_FloatType<"Double", "double"> {
+  let summary = "CIR double-precision float type";
+  let description = [{
+    Floating-point type that represents the `double` type in C/C++. Its
+    underlying floating-point format is the IEEE-754 binar64 format.
+  }];
+}
+
+// Constraints
+
+def CIR_AnyFloat: Type<
+    CPred<"$_self.isa<::mlir::FloatType, ::mlir::cir::FloatType>()">>;
```

Comment:
I see both here and elsewhere checks for accepting both `::mlir::FloatType` and `::mlir::cir::FloatType`. This PR should remove and forbid all uses of `::mlir::FloatType`. You may wonder about f80...

Well, CIR only supports f32 and f64, you can ignore or comment out (with FIXME's) for f80 now, it's not really tested or claimed support (probably got added for consistency but it's just getting in the way...). In follow up PRs, when real support is added for f80, we'll add a proper type for it and add all the bits that matter.

Reply:
Updated. Removed all old `mlir::FloatType` and `mlir::FloatAttr` usages. Some tests that contain yet-not-supported floating-point formats are temporary commented out.

Reply:
Thansk! How many of those you ended up finding? Is it easy to move them over some other test files and XFAIL them so we don't miss testing for the existing relevant stuff in the meantime?

Reply:
> How many of those you ended up finding?

Here is the complete list of test files affected:

- [clang/test/CIR/CodeGen/builtin-floating-point.c](https://github.com/llvm/clangir/pull/385/files#diff-42e4c2418f49dac01cead025aac4b6d2c20e6313ce55eec9f1320a9f8bcc8e92)
- [clang/test/CIR/CodeGen/types.c](https://github.com/llvm/clangir/pull/385/files#diff-489e62a365b5466ef905ff5b853c9618dd7503918f711c79c8067947bc4ab256)
- [clang/test/CIR/Lowering/float.cir](https://github.com/llvm/clangir/pull/385/files#diff-489e62a365b5466ef905ff5b853c9618dd7503918f711c79c8067947bc4ab256)

> Is it easy to move them over some other test files and XFAIL them

Yes we could do this. Although all these tests are not specifically designed for a specific floating point type so it might be a bit strange at first. But we can move these tests back to where they are now after we have supported all possible floating point types.

Reply:
I just looked at those and your current approach is great, thanks!